src/azure-cli/azure/cli/command_modules/acs/_validators.py

```diff
@@ -94,13 +94,13 @@ def validate_k8s_version(namespace):
     """Validates a string as a possible Kubernetes version. An empty string is also valid, which tells the server
     to use its default version."""
     if namespace.kubernetes_version:
-        k8s_release_regex = re.compile(r'^[v|V]?(\d+\.\d+\.\d+.*)$')
```

Comment:
Is reducing the `.*` expected?

Reply:
yes, this is expected. Actually, the original regex has some issues. With .*, some version like 1.2.3.4 is allowed, but we only allow major.minor.patch, which is 1.2.3 is allowed. And our new regex will allow both 1.2 or 1.2.3, but 1.2.3.4 is not allowed.